pkg/appAPIs/appAPIs.go

```diff
@@ -129,15 +131,60 @@ func (cli_api *AppAPI) createAppAPI(createInfo string, token string) ([]byte, er
 
 // To get all the apps information.
 func CreateApp(name string, image string, username string, password string,
-	env []string, port string, token string) error {
+	env []string, envFilePath string, port string, token string) error {
 	// Endpoint to list apps.
 	url := fmt.Sprintf(constants.APPURL)
 	var createInfo string
-	if env != nil {
+
+	/*
+		Following are the cases for env passed through -envPath or -env flag or both
+		1. When environment variable is passed from both -env and -envPath:
+			1.1. If a common key is found in -env and -envPath, an error will occur.
+			1.2. If different env variables are passed in -env and -envPath, a union of both will be considered.
+		2. When environment variable is passed from -env (-e), a string slice will be generated from getEnvSlice function.
+		3. When environment variable is passed from -envPath (-f) (assuming that the file path is valid),
+		   a string slice  will be generated from GetSliceFromEnvFile function.
+	*/
+
+	if env != nil && envFilePath != "" {
+		sliceFromEnv, sliceMap := genEnvSlice(env)
+		sliceFromEnvFile, envMap, err := GetSliceFromEnvFile(envFilePath)
+
+		if err != nil {
+			return fmt.Errorf("%s", err)
+		}
+
+		// If the same key is found in both the .env file passed through --envPath and the --env flag, an error will occur.
+		for key, _ := range envMap {
+			_, found := sliceMap[key]
+			if found {
+				return fmt.Errorf("Duplicate environment variable: %v found. Either remove it from env file or from command line.", key)
+			}
+		}
+
+		slice := append(sliceFromEnv, ",")
+		slice = append(slice, sliceFromEnvFile...)
+
+		if port != "" {
+			createInfo = fmt.Sprintf(`{"name":"%s", "image":"%s", "username":"%s", "password":"%s", "port": "%s", "envs": %v}`, name, image, username, password, port, slice)
+		} else {
+			createInfo = fmt.Sprintf(`{"name":"%s", "image":"%s", "username":"%s", "password":"%s", "envs": %v}`, name, image, username, password, slice)
+		}
+	} else if env != nil {
+		envSlice, _ := genEnvSlice(env)
 		if port != "" {
-			createInfo = fmt.Sprintf(`{"name":"%s", "image":"%s", "username":"%s", "password":"%s", "port": "%s", "envs": %v}`, name, image, username, password, port, genEnvSlice(env))
+			createInfo = fmt.Sprintf(`{"name":"%s", "image":"%s", "username":"%s", "password":"%s", "port": "%s", "envs": %v}`, name, image, username, password, port, envSlice)
+		} else {
+			createInfo = fmt.Sprintf(`{"name":"%s", "image":"%s", "username":"%s", "password":"%s", "envs": %v}`, name, image, username, password, envSlice)
+		}
+	} else if envFilePath != "" {
+		envSlice, _, err := GetSliceFromEnvFile(envFilePath)
+		if err != nil {
+			return fmt.Errorf("%s", err)
+		} else if port != "" {
+			createInfo = fmt.Sprintf(`{"name":"%s", "image":"%s", "username":"%s", "password":"%s", "port": "%s", "envs": %v}`, name, image, username, password, port, envSlice)
 		} else {
```

Comment:
This piece of code has become effectively unreadable.
Could you place it in smaller functions for the reader of the code to make sense out of it ?